File: Data Structures & Algorithms/max-consecutive-ones/submission-0.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        max_cnt=0
        cnt=0
        for i in nums:
        
            if i==1:
                cnt+=1
            else:
                if(cnt>max_cnt):
                    max_cnt=cnt
                    cnt=0
                else:
                    cnt=0
        if cnt > max_cnt:
            max_cnt = cnt
        return max_cnt

File: Data Structures & Algorithms/permutations/submission-0.java
class Solution {
    public List<List<Integer>> permute(int[] nums) {
        List<List<Integer>> res = new ArrayList<>();
        List<Integer> sub = new ArrayList<>();
        boolean used[] = new boolean[nums.length];

        backtrack(res, sub, used, nums);
        return res;
    }

    void backtrack(List<List<Integer>> res, List<Integer> sub, boolean used[],  int nums[]){

        if(nums.length == sub.size()){
            res.add(new ArrayList<>(sub));
            return;
        }

        for(int i=0;i<nums.length;i++){
                if(used[i]) continue;
                sub.add(nums[i]);
                used[i] = true;
                backtrack(res, sub, used, nums);
                used[i] = false;
                sub.remove(sub.size()-1);
            
        }

    }
}
